integration-cli/docker_cli_rm_test.go

```diff
@@ -93,6 +95,17 @@ func (s *DockerSuite) TestRmRestartingContainer(c *check.C) {
 	dockerCmd(c, "run", "--name", name, "--restart=always", "busybox", "date")
 
 	res, _, err := dockerCmdWithError("rm", name)
+
+	for strings.Contains(res, "cannot remove a running container") {
+		// The `date` command hasn't exited yet. It should end
+		// up in a "restarting" state if we wait a bit, though
+		// it is possible that it might be running again by
+		// that time. The wait period between each restart
+		// increases though so we just loop in this condition.
+		time.Sleep(100 * time.Millisecond)
```

Comment:
I am afraid `time.Sleep` isn't good enough.
What will happen if after the 100 msecs the date will be restarted and the container will be running. The test will be still running.